controllers/drpolicy_controller_test.go

```diff
@@ -369,9 +358,18 @@ var _ = Describe("DrpolicyController", func() {
 		configMapUpdate()
 		for i := range manifestWorks {
 			manifestWork := &manifestWorks[i]
-			manifestCount := len(manifestWork.Spec.Workload.Manifests)
-			drClusterManifestWorkUpdateExpect(manifestWork)
-			Expect(len(manifestWork.Spec.Workload.Manifests)).To(BeNumerically(comparator, manifestCount))
+			expectedCount := 2
```

Comment:
This fix had a few root causes:
- We used to check the manifest once its generation changed
- At times a generation change did not change the manifest count, but the manifest count changed only post 2 generations
- The generation changed twice as there was a reconcile for the policy based on the previous test
- The generation should ideally not change as [DeepEqual](https://github.com/RamenDR/ramen/blob/f3df19808d858bd53e905989123fe627ce788757/controllers/util/mw_util.go#L304) should have determined no change to the manifest
- But, the DeepEqual fails as the manifest found and the manifest generated differ, where the generated manifest always contains `"creationTimestamp":null` in `Object.metadata`
- This comes from the meta [DeepCopyInto](https://github.com/kubernetes/apimachinery/blob/78994e499d98ab90598d02007916947c8090eabf/pkg/apis/meta/v1/zz_generated.deepcopy.go#L663)

I was not able to avoid generating the extra null parameter, it does not impact functionality but causes a call to update to the manifest.

The fix hence is to check till the manifest count reaches the desired count.